src/coreclr/jit/lclmorph.cpp

```diff
@@ -1060,6 +1061,22 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 break;
             }
 
+            case GT_NULLCHECK:
+            {
+                assert(TopValue(1).Node() == node);
+                assert(TopValue(0).Node() == node->AsOp()->gtOp1);
+                Value& op = TopValue(0);
+                if (op.IsAddress())
+                {
+                    JITDUMP("Bashing nullcheck of local [%06u] to NOP\n", m_compiler->dspTreeID(node));
+                    node->gtBashToNOP();
+                    m_stmtModified = true;
+                }
+                INDEBUG(TopValue(0).Consume());
+                PopValue();
```

Comment:
This has to call `EscapeValue` in the non-transformed case (that might be the reason for the asserts)